### ISSUE TITLE
dwell time issues

### ISSUE THREAD

**Issue by koch-t** [bliksemlabs/rrrr#101]
I can add to this that the backward search is also going wrong, as the train arrives at Leiden at 11:13.
So we can pinpoint this issue directly to the point where it's fetching times when already boarded a trip.

I've added some notes below in-line

Sent from my Galaxy Nexus ;)

On Oct 30, 2013 1:41 AM, "Thomas" thomaskoch@gmail.com wrote:

> I noticed another bug/artifact. In the backward search of the 2nd itinerary:
> 
> RAIL 10879   1 38925 39552 10:10:00 11:13:00 +0.0 ;NS;Intercity;Den Haag Centraal;Intercity;Lelystad Centrum;Leiden Centraal;

Arrival time is used for boarding stop instead of departure search. Backward search thus happend on board vehicle. ( not to be confused with the onboard trip request functionality)

> It says departure from Lelystad @ 10:10. 10:10 is the arrivaltime at Lelystad, dwelltime is one minuten and departuretime is 10:11. In the forward search this is correctly done so no dataissue:
> 
> RAIL 10879   1 38925 39552 10:11:00 11:15:00 +0.0 ;NS;Intercity;Den Haag Centraal;Intercity;Lelystad Centrum;Leiden Centraal;

Departuretime is used for alighting stop. Forward thus happened on board vehicle.

> Notice that also the arrival time at Leiden is different even though tripid, stopid's are the same.
> Now i'm wondering if this is just a rendering issue or the cause of asymmetric reversals we're seeing.
> 
> TL;DR  Arrival renders as departure time and vice versa. Could be cause asymetric reversals.
> 
> Hi Andrew,
> 
> I've noticed some reversals going wrong, i've attached the used timetable.dat as well, it works with the split_routedesc_dst branch
> 
> Firstly
> in this advice it can't find a trip in the reversal.
> 
>  ./testerrrr --from-idx=4199 --to-idx=17090 -d -D 2013-10-29T12:11:00 -v
> -- Router Request --
> from:  Lelystad, Station Centrum [4199]
> to:    Leidschendam, Rozenrust [17090]
> time:  12:11:00 [32565]
> speed: 1.500000 m/sec
> arrive-by: false
> max xfers: 5
> max time:     --  
> mode: transit
> 
> ITIN 2 rides 
> WALK    -2  -2  4199 38924 12:11:00 12:11:52 +0.0 ;;walk;walk;;Lelystad, Station Centrum;Lelystad Centrum;
> RAIL 12366  13 38924 39551 12:15:00 13:35:00 +0.0 ;NS;Intercity;Vlissingen;Intercity;Lelystad Centrum;Leiden Centraal;
> WALK    -2  -2 39551 17418 13:35:00 13:49:24 +0.0 ;;walk;walk;;Leiden Centraal;Leiden, Breestraat;
> BUS  2105  41 17418 17090 13:50:00 14:12:00 +0.0 ;Veolia;45;Den Haag via Voorschoten;;Leiden, Breestraat;Leidschendam, Rozenrust;
> WAIT    -2  -2 17090 17090 14:12:00 14:12:00 +0.0 ;;walk;walk;;Leidschendam, Rozenrust;Leidschendam, Rozenrust;
> 
> ITIN 3 rides 
> WALK    -2  -2  4199 38924 12:11:00 12:11:52 +0.0 ;;walk;walk;;Lelystad, Station Centrum;Lelystad Centrum;
> RAIL 12366  13 38924 38833 12:15:00 13:44:00 +0.0 ;NS;Intercity;Vlissingen;Intercity;Lelystad Centrum;Den Haag Laan v NOI;
> WALK    -2  -2 38833  9795 13:44:00 13:44:20 +0.0 ;;walk;walk;;Den Haag Laan v NOI;Den Haag, Laan van NOI;
> TRAM 14782 236  9795  9579 13:44:28 13:48:28 +0.0 ;HTM;4;Zoetermeer Javalaan;Tram;Den Haag, Laan van NOI;Voorburg, Leidschendam-Voorburg;
> WALK    -2  -2  9579 17696 13:48:28 13:52:52 +0.0 ;;walk;walk;;Voorburg, Leidschendam-Voorburg;Voorburg, Sijtwendetunnel;
> BUS  2089  42 17696 17091 13:54:00 13:59:00 +0.0 ;Veolia;45;Leiden via Voorschoten;;Voorburg, Sijtwendetunnel;Leidschendam, Rozenrust;
> WALK    -2  -2 17091 17090 13:59:00 13:59:08 +0.0 ;;walk;walk;;Leidschendam, Rozenrust;Leidschendam, Rozenrust;
> Repeated search with reversed request: 
> -- Router Request --
> from:  Lelystad, Station Centrum [4199]
> to:    Leidschendam, Rozenrust [17090]
> time:  13:59:08 [34187]
> speed: 1.500000 m/sec
> arrive-by: true
> max xfers: 2
> max time:  12:11:00
> mode: transit
> Repeated search with reversed request: 
> -- Router Request --
> from:  Lelystad, Station Centrum [4199]
> to:    Leidschendam, Rozenrust [17090]
> time:  13:59:08 [34187]
> speed: 1.500000 m/sec
> arrive-by: true
> max xfers: 2
> max time:  12:11:00
> mode: transit
> ---
> 
> Different example, these are two requests, both from/to the same stop. If i request arrive by @ 12:11 it will arrive @ 11:57. If i want to arrive at 11:56 it will arive at 12:03... Not sure what's going on there, but it looks like the non-symetric reversal issue again.
> 
> -- Router Request --
> from:  Lelystad, Station Centrum [4199]
> to:    Leidschendam, Rozenrust [17090]
> time:  12:11:00 [32565]
> speed: 1.500000 m/sec
> arrive-by: true
> max xfers: 5
> max time:     --  
> mode: transit
> 
> ITIN 2 rides 
> WALK    -2  -2  4199 38925 10:08:56 10:10:00 +0.0 ;;walk;walk;;Lelystad, Station Centrum;Lelystad Centrum;
> RAIL 10879   1 38925 39552 10:10:00 11:13:00 +0.0 ;NS;Intercity;Den Haag Centraal;Intercity;Lelystad Centrum;Leiden Centraal;
> WALK    -2  -2 39552  7281 11:29:32 11:32:00 +0.0 ;;walk;walk;;Leiden Centraal;Leiden, Centraal Station;
> BUS  4771  40  7281 50219 11:32:00 11:57:00 +0.0 ;Arriva;45;Den Haag CS;;Leiden, Centraal Station;Leidschendam, Rozenrust;
> WALK    -2  -2 50219 17090 12:11:00 12:11:00 +0.0 ;;walk;walk;;Leidschendam, Rozenrust;Leidschendam, Rozenrust;
> Repeated search with reversed request: 
> -- Router Request --
> from:  Lelystad, Station Centrum [4199]
> to:    Leidschendam, Rozenrust [17090]
> time:  10:08:56 [30734]
> speed: 1.500000 m/sec
> arrive-by: false
> max xfers: 1
> max time:  12:11:00
> mode: transit
> 
> ITIN 2 rides 
> WALK    -2  -2  4199 38925 10:08:56 10:10:00 +0.0 ;;walk;walk;;Lelystad, Station Centrum;Lelystad Centrum;
> RAIL 10879   1 38925 39552 10:11:00 11:15:00 +0.0 ;NS;Intercity;Den Haag Centraal;Intercity;Lelystad Centrum;Leiden Centraal;
> WALK    -2  -2 39552 17042 11:15:00 11:34:44 +0.0 ;;walk;walk;;Leiden Centraal;Leiden, Korevaarstraat;
> BUS  2105  26 17042 17090 11:37:00 11:57:00 +0.0 ;Veolia;45;Den Haag via Voorschoten;;Leiden, Korevaarstraat;Leidschendam, Rozenrust;
> WAIT    -2  -2 17090 17090 11:57:00 11:57:00 +0.0 ;;walk;walk;;Leidschendam, Rozenrust;Leidschendam, Rozenrust;
> 
> thomas@openkvk /home/projects/bliksem/rrrr $ ./testerrrr --from-idx=4199 --to-idx=17090 -a -D 2013-10-29T11:56:00 -v
> -- Router Request --
> from:  Lelystad, Station Centrum [4199]
> to:    Leidschendam, Rozenrust [17090]
> time:  11:56:00 [32340]
> speed: 1.500000 m/sec
> arrive-by: true
> max xfers: 5
> max time:     --  
> mode: transit
> 
> ITIN 2 rides 
> WALK    -2  -2  4199 38925 10:08:56 10:10:00 +0.0 ;;walk;walk;;Lelystad, Station Centrum;Lelystad Centrum;
> RAIL 10879   1 38925 39552 10:10:00 11:13:00 +0.0 ;NS;Intercity;Den Haag Centraal;Intercity;Lelystad Centrum;Leiden Centraal;
> WALK    -2  -2 39552  7281 11:14:32 11:17:00 +0.0 ;;walk;walk;;Leiden Centraal;Leiden, Centraal Station;
> BUS  4771  38  7281 50219 11:17:00 11:42:00 +0.0 ;Arriva;45;Den Haag CS;;Leiden, Centraal Station;Leidschendam, Rozenrust;
> WALK    -2  -2 50219 17090 11:56:00 11:56:00 +0.0 ;;walk;walk;;Leidschendam, Rozenrust;Leidschendam, Rozenrust;
> Repeated search with reversed request: 
> -- Router Request --
> from:  Lelystad, Station Centrum [4199]
> to:    Leidschendam, Rozenrust [17090]
> time:  10:08:56 [30734]
> speed: 1.500000 m/sec
> arrive-by: false
> max xfers: 1
> max time:  11:56:00
> mode: transit
> 
> ITIN 2 rides 
> WALK    -2  -2  4199 38925 10:08:56 10:10:00 +0.0 ;;walk;walk;;Lelystad, Station Centrum;Lelystad Centrum;
> RAIL 10879   1 38925 39552 10:11:00 11:15:00 +0.0 ;NS;Intercity;Den Haag Centraal;Intercity;Lelystad Centrum;Leiden Centraal;
> WALK    -2  -2 39552 17042 11:15:00 11:34:44 +0.0 ;;walk;walk;;Leiden Centraal;Leiden, Korevaarstraat;
> BUS  2105  26 17042 17107 11:37:00 11:54:00 +0.0 ;Veolia;45;Den Haag via Voorschoten;;Leiden, Korevaarstraat;Leidschendam, GGZ Haagstreek;
> WALK    -2  -2 17107 17090 11:54:00 12:03:36 +0.0 ;;walk;walk;;Leidschendam, GGZ Haagstreek;Leidschendam, Rozenrust;
